src/gleam/string.gleam

```diff
@@ -96,39 +96,94 @@ pub external fn compare(String, String) -> order.Order =
   "gleam_stdlib" "compare_strings"
 
 /// ## Get Substrings
-
-// TODO
-/// Take a substring given a start and end Grapheme indexes. Negative indexes
-/// are taken starting from the *end* of the list.
+///
+/// Take a substring given a starting index and length.
+/// A negative start or length will return an empty string.
+/// Works on Unicode Graphemes.
 ///
 /// ## Examples
 /// ```gleam
-/// slice("gleam", from: 1, to: 3) == "lea"
-/// slice("gleam", from: 1, to: 10) == "leam"
-/// slice("snakes on a plane!", from: -6, to: -1) == "plane"
+/// slice("gleam", from: 1, length: 3) == "lea"
+/// slice("gleam", from: 1, length: 10) == "leam"
+/// slice("snakes on a plane!", from: -6, to: -1) == ""
 /// ```
 ///
-// pub fn slice(out_of string: String, from start: Int, end: Int) -> String {}
+external fn erl_string_slice(string, start, length) -> String =
```

Comment:
Can we have this before the doc comment please? Currently it is attached to the incorrect function.